## Assignment 1 

Suppose you have an innovative product that you have been selling for the past year. You have decided to apply for a patent to protect your IP, but you also did that in order to increase sales as you are aware that businesses deem a patented product more worthy.
Let's make the following assumptions

There is a 50% chance that your product gets patented
* If it does get patented, your sales go up by 25% - 75%, with 50% being the most likely case.
* Without a patent you expect to sell between 1 - 9 million next year, with 3 million being the most likely case.
* We do not have to consider any costs or expenses
    
Suppose a wholesaler offers to buy your entire production and inventory for the year for $6 million
(you won’t be able to sell anything else), would you accept the offer?

In [19]:
import numpy as np
# Specify number of monte carlo simulations
N_ROUNDS = 50
results = []
count=0
print ('Count\t|Patent\t|base sales| Markup \t| Sales|')
print('--------------------------------------------------')

for rnd in range(N_ROUNDS):
    count+=1
    prob_patent = np.random.randint(0, 2)
    base_sales = round(np.random.triangular(1e6, 3e6, 9e6)
        / 1000000, 2)
    patent_markup = np.random.triangular(0.25, 0.5, 0.75)
    sales = round(base_sales + (base_sales 
        * prob_patent * patent_markup), 2)
    print("%i \t| %i \t| %1.6f |\t %1.6f| \t %1.6f"%(count,prob_patent,base_sales,patent_markup,sales))
    results.append(sales)

Count	|Patent	|base sales| Markup 	| Sales|
--------------------------------------------------
1 	| 0 	| 3.470000 |	 0.359762| 	 3.470000
2 	| 1 	| 5.380000 |	 0.420416| 	 7.640000
3 	| 0 	| 4.040000 |	 0.551159| 	 4.040000
4 	| 1 	| 4.870000 |	 0.469834| 	 7.160000
5 	| 0 	| 4.220000 |	 0.644036| 	 4.220000
6 	| 0 	| 6.420000 |	 0.581933| 	 6.420000
7 	| 0 	| 7.090000 |	 0.690954| 	 7.090000
8 	| 1 	| 6.000000 |	 0.541646| 	 9.250000
9 	| 1 	| 2.900000 |	 0.700128| 	 4.930000
10 	| 1 	| 4.430000 |	 0.484523| 	 6.580000
11 	| 0 	| 3.740000 |	 0.361586| 	 3.740000
12 	| 0 	| 3.180000 |	 0.524682| 	 3.180000
13 	| 1 	| 6.230000 |	 0.358001| 	 8.460000
14 	| 1 	| 3.740000 |	 0.621409| 	 6.060000
15 	| 1 	| 2.730000 |	 0.519644| 	 4.150000
16 	| 1 	| 4.350000 |	 0.596166| 	 6.940000
17 	| 0 	| 6.040000 |	 0.567397| 	 6.040000
18 	| 0 	| 7.820000 |	 0.497757| 	 7.820000
19 	| 1 	| 2.420000 |	 0.682882| 	 4.070000
20 	| 1 	| 8.450000 |	 0.527017| 	 12.900000
21 	| 0 	| 2.190000 |	 0.464567| 

Count	|Patent	|base sales| Markup 	| Sales|
-----------|----|-----------|-----------|-------------
1 	| 0 	| 3.470000 |	 0.359762| 	 3.470000
2 	| 1 	| 5.380000 |	 0.420416| 	 7.640000
3 	| 0 	| 4.040000 |	 0.551159| 	 4.040000
4 	| 1 	| 4.870000 |	 0.469834| 	 7.160000
5 	| 0 	| 4.220000 |	 0.644036| 	 4.220000
6 	| 0 	| 6.420000 |	 0.581933| 	 6.420000
7 	| 0 	| 7.090000 |	 0.690954| 	 7.090000
8 	| 1 	| 6.000000 |	 0.541646| 	 9.250000
9 	| 1 	| 2.900000 |	 0.700128| 	 4.930000
10 	| 1 	| 4.430000 |	 0.484523| 	 6.580000
11 	| 0 	| 3.740000 |	 0.361586| 	 3.740000
12 	| 0 	| 3.180000 |	 0.524682| 	 3.180000
13 	| 1 	| 6.230000 |	 0.358001| 	 8.460000
14 	| 1 	| 3.740000 |	 0.621409| 	 6.060000
15 	| 1 	| 2.730000 |	 0.519644| 	 4.150000
16 	| 1 	| 4.350000 |	 0.596166| 	 6.940000
17 	| 0 	| 6.040000 |	 0.567397| 	 6.040000
18 	| 0 	| 7.820000 |	 0.497757| 	 7.820000
19 	| 1 	| 2.420000 |	 0.682882| 	 4.070000
20 	| 1 	| 8.450000 |	 0.527017| 	 12.900000
21 	| 0 	| 2.190000 |	 0.464567| 	 2.190000
22 	| 0 	| 2.860000 |	 0.524290| 	 2.860000
23 	| 0 	| 8.050000 |	 0.511012| 	 8.050000
24 	| 1 	| 6.270000 |	 0.465292| 	 9.190000
25 	| 0 	| 4.920000 |	 0.469111| 	 4.920000
26 	| 1 	| 7.220000 |	 0.440685| 	 10.400000
27 	| 1 	| 5.980000 |	 0.478920| 	 8.840000
28 	| 0 	| 3.810000 |	 0.482395| 	 3.810000
29 	| 1 	| 5.530000 |	 0.604313| 	 8.870000
30 	| 1 	| 2.920000 |	 0.429142| 	 4.170000
31 	| 0 	| 3.610000 |	 0.412015| 	 3.610000
32 	| 0 	| 3.340000 |	 0.411706| 	 3.340000
33 	| 1 	| 4.900000 |	 0.539257| 	 7.540000
34 	| 0 	| 4.690000 |	 0.641628| 	 4.690000
35 	| 0 	| 4.500000 |	 0.500466| 	 4.500000
36 	| 0 	| 2.860000 |	 0.313149| 	 2.860000
37 	| 0 	| 2.160000 |	 0.611192| 	 2.160000
38 	| 1 	| 7.010000 |	 0.558098| 	 10.920000
39 	| 1 	| 5.880000 |	 0.499919| 	 8.820000
40 	| 0 	| 4.090000 |	 0.265641| 	 4.090000
41 	| 0 	| 6.510000 |	 0.536494| 	 6.510000
42 	| 1 	| 5.410000 |	 0.430903| 	 7.740000
43 	| 1 	| 3.590000 |	 0.540065| 	 5.530000
44 	| 0 	| 3.010000 |	 0.485524| 	 3.010000
45 	| 1 	| 5.380000 |	 0.378852| 	 7.420000
46 	| 1 	| 6.690000 |	 0.368614| 	 9.160000
47 	| 0 	| 6.940000 |	 0.309653| 	 6.940000
48 	| 0 	| 6.300000 |	 0.547513| 	 6.300000
49 	| 1 	| 3.060000 |	 0.659056| 	 5.080000
50 	| 1 	| 3.760000 |	 0.497636| 	 5.630000


* Average sales we would expect if we don't accept the wholesaler's deal


* The probability of selling more than 6 million (offered deal)


* Is it a good deal. 